src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+            var returnExpression = new List<SyntaxNode>
+            {
+                generator.ReturnStatement(expression.WithoutTrailingTrivia())
+            };
+
+            var typeSymbol = semanticDocument.SemanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var methodDeclaration = generator.MethodDeclaration(newMethod, returnExpression).WithAdditionalAnnotations(Formatter.Annotation);
+            editor.InsertBefore(methodExpression, methodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter
+        /// </summary>
+        /// 
+        private async Task IntroduceNewMethodOverloadAsync(SemanticDocument semanticDocument,
+                                                           TExpressionSyntax expression,
+                                                           IMethodSymbol methodSymbol,
+                                                           SyntaxEditor editor,
+                                                           CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), true)!;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters.As<IParameterSymbol>());
+            arguments = AddExpressionArgumentToArgumentList(arguments, expression.WithoutTrailingTrivia());
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            List<SyntaxNode> invocationReturn;
+
+            if (!methodSymbol.ReturnsVoid)
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    generator.ReturnStatement(invocation)
+                };
+            }
+            else
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    invocation
+                };
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+
+            var methodDeclaration = generator.MethodDeclaration(newMethod, invocationReturn);
```

Comment:
Perhaps parts of this method could be combined with the above? For example this one seems to lack formatter annotations.

Reply:
Oh the formatter annotations were actually left in on accident, they do not do anything.

Reply:
yes.  this seems odd.  gode gen factory should allow you to make a method that has statements in it.  that's a supported use case.

Reply:
^The code you sent me never gets hit in either the CodeGenerationSymbolFactory or the generator code